vc-worker/src/main/java/com/vmware/flowgate/vcworker/scheduler/job/VCDataService.java

```diff
@@ -168,6 +207,315 @@ public void checkAndUpdateIntegrationStatus(SDDCSoftwareConfig vc, String messag
+                  
+                  restClient.saveAssets(hostMappingAsset);
+               }
+            }
+         }
+
+      } catch (ConnectionException e1) {
+         checkAndUpdateIntegrationStatus(vc, e1.getMessage());
+         return;
+      } catch (ExecutionException e2) {
+         if (e2.getCause() instanceof InvalidLogin) {
+            logger.error("Failed to push data to " + vc.getServerURL(), e2);
+            checkAndUpdateIntegrationStatus(vc, "Invalid username or password.");
+            return;
+         }
+      } catch (Exception e) {
+         logger.error("Failed to sync the host metadata to VC ", e);
+         return;
+      }
+   }
+   
+   private void feedAssetMetricsFormulars(Map<String, Map<String, String>> metrics, String assetId) {
+
+      Map<String, String> cpuMap = new HashMap<>();
+      cpuMap.put(FlowgateConstant.HOST_CPU_USAGE, assetId);
+      cpuMap.put(FlowgateConstant.HOST_CPU_USED, assetId);
+      metrics.put(FlowgateConstant.HOST_CPU, cpuMap);
+      
+      Map<String, String> memoryMap = new HashMap<>();
+      memoryMap.put(FlowgateConstant.HOST_MEMORY_ACTIVE, assetId);
+      memoryMap.put(FlowgateConstant.HOST_MEMORY_BALLOON, assetId);
+      memoryMap.put(FlowgateConstant.HOST_MEMORY_CONSUMED, assetId);
+      memoryMap.put(FlowgateConstant.HOST_MEMORY_SHARED, assetId);
+      memoryMap.put(FlowgateConstant.HOST_MEMORY_SWAP, assetId);
+      memoryMap.put(FlowgateConstant.HOST_MEMORY_USAGE, assetId);
+      metrics.put(FlowgateConstant.HOST_MEMORY, memoryMap);
+      
+      Map<String, String> storageMap = new HashMap<>();
+      storageMap.put(FlowgateConstant.HOST_STORAGE_IORATEUSAGE, assetId);
+      storageMap.put(FlowgateConstant.HOST_STORAGE_USAGE, assetId);
+      storageMap.put(FlowgateConstant.HOST_STORAGE_USED, assetId);
+      metrics.put(FlowgateConstant.HOST_STORAGE, storageMap);
+   }
+   private Map<String, Long> fetchDataByType(PerformanceManager performanceManager,
+         CounterInfo[] counterInfos, HostSystem host, String groupInfoKey, String unitInfoSummary,
+         String nameInfoSummary) {
+
+      Map<Integer, CounterInfo> counters = new HashMap<Integer, CounterInfo>();
+      for (int i = 0; i < counterInfos.length; i++) {
+         CounterInfo counterInfo = counterInfos[i];
+
+         if (groupInfoKey.equalsIgnoreCase(counterInfo.getGroupInfo().getKey())
+               && unitInfoSummary.equalsIgnoreCase(counterInfo.getUnitInfo().getSummary())
+               && nameInfoSummary.equalsIgnoreCase(counterInfo.getNameInfo().getSummary())) {
+            counters.put(new Integer(counterInfo.getKey()), counterInfo);
+         }
+      }
+      MetricId[] queryAvailableMetric =
+            performanceManager.queryAvailableMetric(host._getRef(), null, null, new Integer(20));
+      ArrayList<MetricId> metricIdList = new ArrayList<MetricId>();
+      if (queryAvailableMetric.length > 0) {
+         for (int i = 0; i < queryAvailableMetric.length; i++) {
```

Comment:
check the metric id format.

Reply:
(vim.MetricId) {
   dynamicType = null,
   dynamicProperty = null,
   counterId = 182,
   instance = d7c13f5a-90cbd366
}